src/metkit/api/metkit_c.h

```diff
@@ -0,0 +1,238 @@
+#ifndef METKIT_API_METKIT_C_H
+#define METKIT_API_METKIT_C_H
+
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * TYPES
+ * -----*/
+
+struct metkit_marsrequest_t;
+typedef struct metkit_marsrequest_t metkit_marsrequest_t;
+
+struct metkit_requestiterator_t;
+/** RequestIterator for iterating over vector of Request instances */
+typedef struct metkit_requestiterator_t metkit_requestiterator_t;
+
+struct metkit_paramiterator_t;
+/** Iterator for iterating over parameters in Request */
+typedef struct metkit_paramiterator_t metkit_paramiterator_t;
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * ERROR HANDLING
+ * -------------- */
+
+typedef enum metkit_error_values_t
+{
+    METKIT_SUCCESS            = 0, /* Operation succeded. */
+    METKIT_ITERATION_COMPLETE = 1, /* All elements have been returned */
+    METKIT_ERROR              = 2, /* Operation failed. */
+    METKIT_ERROR_UNKNOWN      = 3, /* Failed with an unknown error. */
+    METKIT_ERROR_USER         = 4, /* Failed with an user error. */
+    METKIT_ERROR_ASSERT       = 5  /* Failed with an assert() */
+} metkit_error_enum_t;
+
+const char* metkit_get_error_string(int err);
+
+/* -----------------------------------------------------------------------------
+ * HELPERS
+ * ------- */
+
+/**
+ * @brief Set MetKit version.
+ *
+ * @param version Version string
+ * @return int Error code
+ */
+int metkit_version(const char** version);
+
+/**
+ * @brief Set MetKit git sha1 version.
+ *
+ * @param sha1 SHA1 version string
+ * @return int Error code
+ */
+int metkit_vcs_version(const char** sha1);
+
+/**
+ * @brief Initialise Main() context.
+ *
+ * @note This is ONLY required when Main() is NOT initialised, such as loading
+ * the MetKit as shared library in Python.
+ * @return int Error code
+ */
+int metkit_initialise();
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * PARSING
+ * --- */
+
+/**
+ * Parse MARS requests into RequestIterator of Request instances. Resulting RequestIterator
+ * must be deallocated with metkit_free_requestiterator
+ * @param str MARS requests
+ * @param requests Allocates RequestIterator object
+ * @return int Error code
+ */
+int metkit_parse_marsrequest(const char* str, metkit_requestiterator_t** requests, bool strict = false);
```

Comment:
This returns mars request**s** (plural), but in the function name it is only singular. That isn't cool.

Also, please make sure you run this through a C compiler --- NOT a C++ compiler. Note that C does not support default arguments.

Reply:
And you will need to include <stdbool.h> if you want to use bool - as bool is not a builtin type in C.